common/src/main/scala/db/SqlRegistrationRepository.scala

```diff
@@ -73,4 +76,46 @@ class SqlRegistrationRepository[F[_]: Async](xa: Transactor[F])
       """
       .update.run
 
+  def countPerPlatformForTopics(topics: NonEmptyList[Topic]): F[PlatformCount] = {
+    val q = fr"""
+      SELECT
+        count(1) as total,
+        coalesce(sum(case platform when 'ios' then 1 else 0 end), 0) as ios,
+        coalesce(sum(case platform when 'android' then 1 else 0 end), 0) as android,
+        coalesce(sum(case platform when 'newsstand' then 1 else 0 end), 0) as newsstand
+      FROM
+        (
+          SELECT DISTINCT
+            token,
+            platform
+          FROM
+            registrations
+          WHERE
+      """ ++
+        Fragments.in(fr"topic", topics) ++
+    fr"""
+        ) as distinct_registrations
+    """
+
+    q.query[PlatformCount]
+      .unique
+      .transact(xa)
+  }
+
+  override def countPerPlatformForTopic(topic: Topic): F[PlatformCount] = {
+    sql"""
+      SELECT
+        count(1) as total,
+        coalesce(sum(case platform when 'ios' then 1 else 0 end), 0) as ios,
+        coalesce(sum(case platform when 'android' then 1 else 0 end), 0) as android,
+        coalesce(sum(case platform when 'newsstand' then 1 else 0 end), 0) as newsstand
+      FROM
+        registrations
+      WHERE
+        topic = $topic
+      """
+      .query[PlatformCount]
+      .unique
+      .transact(xa)
+  }
```

Comment:
As discussed face to face, this is a nice optimization but it also means maintaining to methods/queries. 
Let's start with a single generic one `countPerPlatformForTopics` and see if the cost worth adding more code later